bcftbx/cmdparse.py

```diff
@@ -234,90 +242,104 @@ def print_command(self,cmd,message=None):
 # Functions
 #######################################################################
 
-def add_nprocessors_option(parser,default_nprocessors,default_display=None):
+def add_nprocessors_option(parser, default_nprocessors, default_display=None):
     """Add a '--nprocessors' option to a parser
-    
-    Given an OptionParser 'parser', add a '--nprocessors' option.
+
+    Given an ArgumentParser 'parser', add a '--nprocessors' option.
 
     The value of this option can be accessed via the 'nprocessors'
     attribute of the parser options.
 
     If 'default_display' is not None then this value will be shown
     in the help text, rather than the value supplied for the default.
 
-    Returns the input OptionParser object.
+    Returns the input ArgumentParser object.
 
     """
     if default_display is None:
         default_display = default_nprocessors
-    parser.add_option('--nprocessors',action='store',
-                      dest='nprocessors',default=default_nprocessors,
-                      help="explicitly specify number of processors/cores to use "
-                      "(default %s)" % 
-                      default_display)
+    if isinstance(parser, argparse.ArgumentParser):
+        add_cmd = parser.add_argument
+    else: add_cmd = parser.add_option
```

Comment:
It's preferable to split the clause following the `else` statement onto a separate new line, i.e.

        if isinstance(parser, argparse.ArgumentParser):
           add_cmd = parser.add_argument
        else:
           add_cmd = parser.add_option